extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -732,7 +789,89 @@ export class SchemaCompareResult {
+				this.startCompare();
+			}
+		});
+	}
+
+	private createOpenScmpButton(view: azdata.ModelView) {
+		this.openScmpButton = view.modelBuilder.button().withProperties({
+			label: localize('schemaCompare.openScmpButton', 'Open .scmp file'),
+			iconPath: {
+				light: path.join(__dirname, 'media', 'open-scmp.svg'),
+				dark: path.join(__dirname, 'media', 'open-scmp-inverse.svg')
+			},
+			title: localize('schemaCompare.openScmpButtonTitle', 'Load source, target, and options saved in an .scmp file')
+		}).component();
+
+		this.openScmpButton.onDidClick(async (click) => {
+			Telemetry.sendTelemetryEvent('SchemaCompareOpenScmpStarted');
+			const rootPath = vscode.workspace.rootPath ? vscode.workspace.rootPath : os.homedir();
+			let fileUris = await vscode.window.showOpenDialog(
+				{
+					canSelectFiles: true,
+					canSelectFolders: false,
+					canSelectMany: false,
+					defaultUri: vscode.Uri.file(rootPath),
+					openLabel: localize('schemaCompare.openFile', 'Open'),
+					filters: {
+						'scmp Files': ['scmp'],
+					}
+				}
+			);
+
+			if (!fileUris || fileUris.length === 0) {
+				return;
+			}
+
+			let fileUri = fileUris[0];
+			const service = await SchemaCompareResult.getService('MSSQL');
+			const result = await service.schemaCompareOpenScmp(fileUri.fsPath);
+			if (!result || !result.success) {
+				Telemetry.sendTelemetryEvent('SchemaCompareOpenScmpFailed', {
+					'errorType': getTelemetryErrorType(result.errorMessage)
+				});
+				vscode.window.showErrorMessage(
+					localize('schemaCompare.openScmpErrorMessage', "Open scmp failed: '{0}'", (result && result.errorMessage) ? result.errorMessage : 'Unknown'));
+				return;
+			}
+
+			if (result.sourceEndpointInfo && result.sourceEndpointInfo.endpointType === azdata.SchemaCompareEndpointType.Database) {
+				// only set endpoint info if able to connect to the database
+				const ownerUri = await verifyConnectionAndGetOwnerUri(result.sourceEndpointInfo);
+				if (ownerUri) {
+					this.sourceEndpointInfo = result.sourceEndpointInfo;
+					this.sourceEndpointInfo.ownerUri = ownerUri;
+				}
+			} else {
+				this.sourceEndpointInfo = result.sourceEndpointInfo;
+			}
+
+			if (result.targetEndpointInfo && result.targetEndpointInfo.endpointType === azdata.SchemaCompareEndpointType.Database) {
+				const ownerUri = await verifyConnectionAndGetOwnerUri(result.targetEndpointInfo);
+				if (ownerUri) {
+					this.targetEndpointInfo = result.targetEndpointInfo;
+					this.targetEndpointInfo.ownerUri = ownerUri;
+				}
+			} else {
+				this.targetEndpointInfo = result.targetEndpointInfo;
+			}
+
+			this.updateSourceAndTarget();
+			this.deploymentOptions = result.deploymentOptions;
+			this.scmpSourceExcludes = result.excludedSourceElements;
+			this.scmpTargetExcludes = result.excludedTargetElements;
+			this.sourceTargetSwitched = result.originalTargetName.toLowerCase() !== this.targetEndpointInfo.databaseName.toLowerCase();
```

Comment:
>result.originalTargetName.toLowerCase() !== this.targetEndpointInfo.databaseName.toLowerCase() [](start = 31, length = 94)

On a case sensitive server, could the toLowerCase() cause problems with databases that have the same name and differ only in case?

Reply:
good point. I'll remove the toLowerCase()s from this comparison